### DISCUSSION TITLE
How to place dynamic plot in flet.

### DISCUSSION THREAD

**Discussion by Mateusz-Krajewski** [flet-dev/flet#2207] [Q&A]
Hi I have app, which receive data every seconds, I want to add chart but they need to be updated every one seconds ,matlplotlib and plotly have option to build dynamic plots but how to place them in flet app?

**Comment by indigo360** [comment]
I have created a custom control class for this which builds upon flet line chart, as the app starts (e.g in main) I initialise a chart instance of type custom line chart with the numbers of Y series to create. As the app updates in a async timed loop (I'm doing every minute) it calls the update chart function in the class to add the new Y into the series datapoints and then updates itself (as all custom controls need their own build)

<img width="877" alt="Screenshot 2023-12-11 at 17 17 22" src="https://github.com/flet-dev/flet/assets/33873235/f1f2f074-6f31-4b66-aa87-9d6ff8fda493">
